src/APL/fileAPL.test.ts

```diff
@@ -0,0 +1,184 @@
+        const stubAuthData = {
+          domain: "example.com",
+          token: "example-token",
+        };
+
+        vi.spyOn(fsPromises, "access").mockResolvedValue();
+        vi.spyOn(fsPromises, "readFile").mockResolvedValue(JSON.stringify(stubAuthData));
+
+        const apl = new FileAPL();
+
+        expect(await apl.get(stubAuthData.domain)).toStrictEqual(stubAuthData);
+      });
+
+      it("Returns undefined for unknown domain", async () => {
+        const stubAuthData = {
+          domain: "example.com",
+          token: "example-token",
+        };
+
+        vi.spyOn(fsPromises, "access").mockResolvedValue();
+        vi.spyOn(fsPromises, "readFile").mockResolvedValue(JSON.stringify(stubAuthData));
+
+        const apl = new FileAPL();
+
+        expect(await apl.get("unknown-domain.example.com")).toBe(undefined);
+      });
+    });
+
+    describe("delete", () => {
+      afterEach(() => {
+        vi.clearAllMocks();
+      });
+
+      it("Should override file when called with known domain", async () => {
+        const stubAuthData = {
+          domain: "example.com",
+          token: "example-token",
+        };
+
+        vi.spyOn(fsPromises, "access").mockResolvedValue();
+        vi.spyOn(fsPromises, "readFile").mockResolvedValue(JSON.stringify(stubAuthData));
+
+        const apl = new FileAPL();
+
+        expect(await apl.get(stubAuthData.domain)).toStrictEqual(stubAuthData);
+      });
+
+      it("Should not delete data when called with unknown domain", async () => {
+        const stubAuthData = {
+          domain: "example.com",
+          token: "example-token",
+        };
+
+        vi.spyOn(fsPromises, "access").mockResolvedValue();
+        vi.spyOn(fsPromises, "readFile").mockResolvedValue(JSON.stringify(stubAuthData));
+
+        const spyWriteFile = vi.spyOn(fsPromises, "writeFile").mockResolvedValue();
+
+        const apl = new FileAPL();
+
+        await apl.delete("unknown-domain.example.com");
+
+        expect(spyWriteFile).toBeCalledTimes(0);
+      });
+    });
+
+    describe("getAll", () => {
+      afterEach(() => {
+        vi.clearAllMocks();
+      });
+
+      it("Should return list with one item when auth data are existing", async () => {
+        const stubAuthData = {
+          domain: "example.com",
+          token: "example-token",
+        };
+
+        vi.spyOn(fsPromises, "access").mockResolvedValue();
+        vi.spyOn(fsPromises, "readFile").mockResolvedValue(JSON.stringify(stubAuthData));
+
+        const apl = new FileAPL();
+
+        expect(await apl.getAll()).toStrictEqual([stubAuthData]);
+      });
+
+      it("Should return empty list when auth data are empty", async () => {
+        vi.spyOn(fsPromises, "access").mockResolvedValue();
+        vi.spyOn(fsPromises, "readFile").mockResolvedValue("{}");
+
+        const apl = new FileAPL();
+
+        expect(await apl.getAll()).toStrictEqual([]);
+      });
+    });
+  });
+
+  describe("FileAPL utils", () => {
```

Comment:
Theres a bit of logic in those functions - handle errors and make sure we return desired structure. Thought it would be better to test edge cases of this function alone. WDYT?

Reply:
AFAIK its a bad pattern and need of testing private methods is also a sign of something wrong. Edge cases can be tested via public methods too. Private method will never be called directly by the client